### ISSUE TITLE
unable to apply citrixadc_servicegroup_servicegroupmember_binding

### ISSUE THREAD

**Issue by hsmade** [citrix/terraform-provider-citrixadc#275]
When trying to apply the `citrixadc_servicegroup_servicegroupmember_binding` (with -target), I get the following error:

```bash
citrixadc_servicegroup_servicegroupmember_binding.my-service-http[0]: Creating...
2021/10/18 15:07:06 [DEBUG] citrixadc_servicegroup_servicegroupmember_binding.my-service-http[0]: applying the planned Create change
2021-10-18T15:07:06.591Z [DEBUG] plugin.terraform-provider-citrixadc_v1.7.0: 2021/10/18 15:07:06 [DEBUG]  citrixadc-provider: In createServicegroup_servicegroupmember_bindingFunc
2021-10-18T15:07:07.076Z [DEBUG] plugin.terraform-provider-citrixadc_v1.7.0: 2021/10/18 15:07:07 [DEBUG] citrixadc-provider:  In readServicegroup_servicegroupmember_bindingFunc
2021-10-18T15:07:07.076Z [DEBUG] plugin.terraform-provider-citrixadc_v1.7.0: 2021/10/18 15:07:07 [DEBUG] citrixadc-provider: Reading servicegroup_servicegroupmember_binding state my-service-http-svc,my-service-http
2021-10-18T15:07:07.419Z [DEBUG] plugin.terraform-provider-citrixadc_v1.7.0: 2021/10/18 15:07:07 [DEBUG] citrixadc-provider: dataArr: [map[customserverid:None dbsttl:1 delay:0 delay1:0 graceful:NO hashid:0 ip:1.2.3.4 nameserver:5.6.7.8 port:31791 serverid:0 servername:ns_server001.node.dc.consul servicegroupname:my-service-http-svc state:ENABLED statechangetimesec:Mon Oct 18 15:41:29 2021 svcitmpriority:1 svrstate:UP tickssincelaststatechange:26 weight:1]]
2021-10-18T15:07:07.419Z [DEBUG] plugin.terraform-provider-citrixadc_v1.7.0: 2021/10/18 15:07:07 [WARN] citrixadc-provider: port is zero
2021-10-18T15:07:07.419Z [DEBUG] plugin.terraform-provider-citrixadc_v1.7.0: 2021/10/18 15:07:07 [WARN] citrixadc-provider: foundIndex: -1
2021-10-18T15:07:07.419Z [DEBUG] plugin.terraform-provider-citrixadc_v1.7.0: 2021/10/18 15:07:07 [DEBUG] citrixadc-provider: FindResourceArrayWithParams monitor name not found in array
2021-10-18T15:07:07.420Z [DEBUG] plugin.terraform-provider-citrixadc_v1.7.0: 2021/10/18 15:07:07 [WARN] citrixadc-provider: Clearing servicegroup_servicegroupmember_binding state my-service-http-svc,my-service-http
2021/10/18 15:07:07 [DEBUG] citrixadc_servicegroup_servicegroupmember_binding.my-service-http[0]: apply errored, but we're indicating that via the Error pointer rather than returning it: Provider produced inconsistent result after apply: When applying changes to citrixadc_servicegroup_servicegroupmember_binding.my-service-http[0], provider "registry.terraform.io/-/citrixadc" produced an unexpected new value for was present, but now absent.

This is a bug in the provider, which should be reported in the provider's own issue tracker.
2021/10/18 15:07:07 [ERROR] <root>: eval: *terraform.EvalSequence, err: Provider produced inconsistent result after apply: When applying changes to citrixadc_servicegroup_servicegroupmember_binding.my-service-http[0], provider "registry.terraform.io/-/citrixadc" produced an unexpected new value for was present, but now absent.

This is a bug in the provider, which should be reported in the provider's own issue tracker.
```

my config:
```hcl
resource "citrixadc_server" "my-service-http" {
  count = var.count
  name = "my-service-http"
  domain = "my-service-http.service.consul"
  querytype = "SRV"
}

resource "citrixadc_lbvserver" "my-service-http-lb" {
  count = signum(var.count)
  name = "my-service-http-lb"
  servicetype = "HTTP"
  port = 80
  ipv46 = var.ip
}

resource "citrixadc_servicegroup" "my-service-http-svc" {
  count = signum(var.count)
  servicegroupname = "my-service-http-svc"
  servicetype = "HTTP"
  autoscale = "DNS"
}

resource "citrixadc_lbvserver_servicegroup_binding" "my-service-http" {
  count = signum(var.count)
  name = citrixadc_lbvserver.my-service-http-lb[0].name
  servicegroupname = citrixadc_servicegroup.my-service-http-svc[0].servicegroupname
}

resource "citrixadc_servicegroup_servicegroupmember_binding" "my-service-http" {
  count = signum(var.count)
  servicegroupname = citrixadc_servicegroup.my-service-http-svc[0].servicegroupname
  servername       = citrixadc_server.my-service-http[0].name
  nameserver       = "5.6.7.8"
  dbsttl = "1" 
}
```

What might be related, is that when I create the binding manually on the citrix device, and try to import it, I get:
```bash
citrixadc_servicegroup_servicegroupmember_binding.my-service-http[0]: Importing from ID "my-service-http."...

Error: resource citrixadc_servicegroup_servicegroupmember_binding doesn't support import
```

Plugin version: 1.5.0 (tried 1.7.0 as well, with the same results)
Terraform version: 0.12.31
citrix version: 13.0

**Comment by hsmade** [comment]
It actually happens here:
![image](https://user-images.githubusercontent.com/1136373/137914082-91a5e3b4-db28-4663-92ca-4e0a0fc6ad69.png)
in eval_apply.go, which does:

```go
	if actual.IsNull() && !planned.IsNull() {
		errs = append(errs, path.NewErrorf("was present, but now absent"))
		return errs
	}
```

So the plugin returns an empty value object in this case

**Comment by andrewhadam** [comment]
I ran into the above today while doing something very similar to the above example.
Citrix Version: 13.82.42
Terraform: 1.0.7
Plugin Version: 1.7.0

I thought for a bit the problems were different because mine actually hits the same error when its using "citrixadc_lbvserver_servicegroup_binding" but found as soon as I commented out the "citrixadc_servicegroup_servicegroupmember_binding" portion of my code, my servers were binding exactly as I would expect them to so something funky definitely happening within the reported module here.

**Comment by giorgos-nikolopoulos** [comment]
Fixed with #278

Please use the new flag `disable_read` in the `citrixadc_servicegroup_servicegroupmember_binding` resource that binds the domain server.

This will disable the read operation which causes the error.

Unfortunately we cannot deduce the actual servicemembers from the NITRO API response.
As long as the local and remote state remain in sync this is not a problem.
If a change is made outside terraform then reconciliation of the remote and local state will have to be manually performed.
